controllers/infinispan_controller.go

```diff
@@ -290,6 +290,32 @@ func (reconciler *InfinispanReconciler) Reconcile(ctx context.Context, ctrlReque
 		}
 	}
 
+	if infinispan.HasSites() {
+		tunnel := &appsv1.Deployment{}
```

Comment:
Whole get/create/update cycle could be replaced with controllerutil.CreateOrUpdate function